textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/source/IndexBasedSourceOperator.java

```diff
@@ -45,6 +56,11 @@ public void close() throws DataFlowException {
 			throw new DataFlowException(e.getMessage(), e);
 		}
 	}
+	
+	public void resetPredicate(IPredicate predicate){
+	    this.predicate = (DataReaderPredicate)predicate;
+	    cursor = CLOSED;

```

Comment:
It's a question about the protocol.  We could "re-open" the cursor so that the caller doesn't need to call "open" again before using it.  I prefer to reset the cursor.  (If so, we may need to delete the new test case written by @sandeepreddy602 ).  No matter, we need to add comments to explain the protocol.